server/leafnode.go

```diff
@@ -649,6 +649,14 @@ func (s *Server) createLeafNode(conn net.Conn, remote *leafNodeCfg) *client {
 			return nil
 		}
 
+		// Not sure that can happen, but in case the connection was marked
+		// as closed during the call to parse...
+		if c.isClosed() {
```

Comment:
Let's make this a small anonymous function inside this function to clean up?

`ifClosedReturn := func() `

Reply:
I would still need to have something like:
```
if closedReturn() {
   return nil
}
```